matchmaking.py

```diff
@@ -25,7 +26,8 @@ def __init__(self, li, config, user_profile):
         self.last_user_profile_update_time = Timer(5 * 60)  # 5 minutes
         self.min_wait_time = 60  # Wait 60 seconds before creating a new challenge to avoid hitting the api rate limits.
         self.challenge_id = None
-        self.block_list = []
+        self.block_list = copy.copy(self.matchmaking_cfg.get("block_list")) or []
+        assert isinstance(self.block_list, list)

```

Comment:
A better place for this check is the `load_config()` function in config.py.

Reply:
I think it would be better to use `self.block_list = (self.matchmaking_cfg.get("block_list") or []).copy()`, so we don't have to import `copy`.